modules/highgui/src/window_cocoa.mm

```diff
@@ -360,19 +322,18 @@ CV_IMPL int cvCreateTrackbar2(const char* trackbar_name,
                               CvTrackbarCallback2 on_notify2,
                               void* userdata)
 {
-    //cout <<"cvCreateTrackbar2" << endl;
-    NSAutoreleasePool* localpool = [[NSAutoreleasePool alloc] init];
-    int res = cvCreateTrackbar(trackbar_name, window_name, val, count, NULL);
-    if(res) {
-        CVWindow *window = cvGetWindow(window_name);
-        if (window && [window respondsToSelector:@selector(sliders)]) {
-            CVSlider *slider = [[window sliders] valueForKey:[NSString stringWithFormat:@"%s", trackbar_name]];
-            [slider setCallback2:on_notify2];
-            [slider setUserData:userdata];
+    @autoreleasepool{
+        int res = cvCreateTrackbar(trackbar_name, window_name, val, count, NULL);
+        if(res) {
+            CVWindow *window = cvGetWindow(window_name);
+            if (window && [window respondsToSelector:@selector(sliders)]) {
+                CVSlider *slider = [[window sliders] valueForKey:[NSString stringWithFormat:@"%s", trackbar_name]];
+                [slider setCallback2:on_notify2];
+                [slider setUserData:userdata];
+            }
         }
+        return res;
```

Comment:
It seems like return is misplaced - it should be outside of the `@autoreleasepool` scope